### PR TITLE
Implement user input history

### LINKED ISSUE #132: As a fast typer with low accuracy, I can retrieve my previous command with a single keystroke
...so that I can correct my typos easily.

### PR DESCRIPTION
**Pull request by jmleong666**
Closes #132.

**New features/changes**
- Implement navigating user input history on the command box. Pressing the up key while on the command box gets the previous input, while pressing the down key gets the next input. If there is no previous or next input, the command box becomes empty.
- Add assertions to the archive add and remove commands.

**Implementations**
- Introduce a new class `UserInputHistory` to store all user inputs within the session. User inputs are stored in a `LinkedList`, and its `ListIterator` is used to navigate between inputs. Whenever the user enters a command, add the input to the `userInputHistory` of the command box and reset its iterator.

**Possible improvements**
- Write user inputs to a storage file so that the user can access inputs from previous sessions.
- Improve input history navigation (Pressing down after up returns the same input by the `ListIterator` navigation logic).

**Testing**
- Add unit tests for the `UserInputHistory` class.

**Demo**
stonksbook_up_down_demo

### CODE REVIEW

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 1]
```diff
@@ -25,16 +28,32 @@
...
+        }
+    }
+
     /**
      * Handles the Enter button pressed event.
      */
     @FXML
     private void handleCommandEntered() {
+        userInputHistory.addToHistory(commandTextField.getText());
         try {
```
Comment (AaronnSeah) [inline src/main/java/seedu/address/ui/CommandBox.java:60]: you might want to consider only adding the text if its not blank?

File: src/main/java/seedu/address/ui/UserInputHistory.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package seedu.address.ui;
```
Comment (AaronnSeah) [inline src/main/java/seedu/address/ui/UserInputHistory.java:1]: will unit tests be added in future PR?

**Review by AaronnSeah — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/ui/UserInputHistory.java (hunk 2 above)
Comment (jmleong666) [reply src/main/java/seedu/address/ui/UserInputHistory.java:1]: Will add them in this PR!

File: src/main/java/seedu/address/ui/CommandBox.java (hunk 1 above)
Comment (jmleong666) [reply src/main/java/seedu/address/ui/CommandBox.java:60]: Yeah should've considered that earlier.

**Pull request merged**